src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -376,12 +376,16 @@ def generate_config(
 
         assert self.TYPE == daemon_spec.daemon_type
         spec = cast(PrometheusSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
-
         try:
             retention_time = spec.retention_time if spec.retention_time else '15d'
         except AttributeError:
             retention_time = '15d'
 
+        try:
+            targets = spec.targets
+        except AttributeError:
+            targets = []
```

Comment:
maybe some warning log could be useful for debugging weird situations? is this expected to happen in normal operations?